src/Aspire.Hosting.Azure.ContainerRegistry/AzureContainerRegistryResource.cs

```diff
@@ -0,0 +1,42 @@
+﻿#pragma warning disable ASPIRECOMPUTE001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Azure.Provisioning.ContainerRegistry;
+using Azure.Provisioning.Primitives;
+
+namespace Aspire.Hosting.Azure.ContainerRegistry;
+
+/// <summary>
+/// Represents an Azure Container Registry resource.
+/// </summary>
+public class AzureContainerRegistryResource(string name, Action<AzureResourceInfrastructure> configureInfrastructure)
+    : AzureProvisioningResource(name, configureInfrastructure), IContainerRegistry
```

Comment:
Do we want IResourceWithConnectionString?

Reply:
I wanted to avoid this until we had the chance to figure out what it means to `WithReference` a container registry from a non-environment resource. Does `IResourceWithConnectionString` influence anything else outside of references?

Reply:
Agree we should avoid it for now.

Reply:
Maybe we should remove the `WithRoleAssignments` API then? I don't think there is a scenario for it if we aren't allowing references to the container registry.